workitem/workitem_repository_blackbox_test.go

```diff
@@ -146,6 +146,28 @@ func (s *workItemRepoBlackBoxTest) TestCreate() {
 		require.Nil(t, wi)
 	})
 
+	s.T().Run("disallow creation if WIT belongs to different spacetemplate", func(t *testing.T) {
+		fxt := tf.NewTestFixture(t, s.DB,
+			tf.SpaceTemplates(2),
+			tf.Spaces(1, func(fxt *tf.TestFixture, idx int) error {
+				fxt.Spaces[idx].SpaceTemplateID = fxt.SpaceTemplates[idx].ID
+				return nil
+			}),
+			tf.WorkItemTypes(2, func(fxt *tf.TestFixture, idx int) error {
+				fxt.WorkItemTypes[idx].SpaceTemplateID = fxt.SpaceTemplates[idx].ID
+				return nil
+			}),
+		)
+		wi, err := s.repo.Create(
+			s.Ctx, fxt.Spaces[0].ID, fxt.WorkItemTypes[1].ID,
+			map[string]interface{}{
+				workitem.SystemTitle: "Some title",
+			}, fxt.Identities[0].ID)
+		require.Error(t, err)
+		require.IsType(t, errors.BadParameterError{}, err)
```

Comment:
The function `func (r *GormWorkItemRepository) Create(...)` has more than one place in which it returns a bad parameter. Can you check that the right error text is also contained within the error, just to be sure?

Reply:
Please see https://github.com/fabric8-services/fabric8-wit/pull/2115/commits/dedbb165140bfa43efcafa98c3908a4c766856c1#diff-1999ee68a49d5d171f3f4b3e1c4ccd04R165